### DISCUSSION TITLE
Authorization by Policy

### DISCUSSION THREAD

**Discussion by Brilbroeder** [graphql-dotnet/graphql-dotnet#3663] [General]
We have a Rest-Api and adding some end-point via graphql. We set up types, queries and schema. Configure it. All works fine.
Next step we want to add authorization by Policy.
We already have auth policies on our rest-api endpoints and that works fine.

From the docs and source this seems the way to go:

```
        public DummyQuery()
        {
            Name = "DummyQuery";
            Field<ListGraphType<DummyType>, List<Dummy>>("Dummies")
                .Resolve(Resolve)
                .AuthorizeWithPolicy("DummyPolicy");

```

but adding .AuthorizeWithPolicy does not change a thing. The endpoint is still accessible if you provide a (valid) bearer token or not.

We have tried to apply a policy in the middleware

                webApplication.UseGraphQL<DummieSchema>(ApiPath, builder =>
                {
                    builder.AuthorizedPolicy ="DummyPolicy";
                });

This does kick in, but does not allow access even with a valid bearer token. I put a breakpoint in our Authorizer, but it does not get hit.

Adding

`                    builder.AuthorizationRequired = true;
`

does not change it for the better.

What do we want to accomplish ?
We want to use the same auth policies we use on the rest-api endpoint to protect the graphql endpoints. 

Any suggestions ?

**Comment by Shane32** [comment]
This is very simple. Please review our extensive documentation on setting up authorization here:

https://github.com/graphql-dotnet/server#authorization-configuration

We also have a number of samples including one specific for JWT bearer tokens within the repository. 

https://github.com/graphql-dotnet/server#samples

Please let me know after you’ve had a chance to review and still are unable to get it working.

**Comment by Shane32** [comment]
If you have multiple authentication schemes set up in ASP.NET Core, then see https://github.com/graphql-dotnet/server/issues/1046

**Comment by Brilbroeder** [comment]
It is very simple indeed. For future reference. 
The missing link : we did not had app.UseAuthentication(); setup in the pipeline.
